core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java

```diff
@@ -508,6 +506,11 @@ public void onTopologyChange(TopologyChangedEvent<?, ?> tce) {
       }
    }
 
+   @ViewChanged
+   public void onViewChange(ViewChangedEvent e) {
+      cleanupLeaverTransactions(e.getNewMembers());
```

Comment:
Can we invoke this in the view change listener?  Wouldn't the leaver transactions require a remote rollback to be sent?  We don't want to run anything that may block during these listeners I would think.